pyomo/gdp/plugins/hull.py

```diff
@@ -648,54 +651,40 @@ def _declare_disaggregated_var_bounds(
         if ub:
             bigmConstraint.add(ub_idx, disaggregatedVar <= ub * var_free_indicator)
 
+        original_var_info = original_var.parent_block().private_data()
+        if 'disaggregated_var_map' not in original_var_info:
+            original_var_info['disaggregated_var_map'] = ComponentMap()
+        disaggregated_var_map = original_var_info['disaggregated_var_map']
```

Comment:
We should add a `default=` or `initialize=` argument to `private_data()` that the method can use to initialize the private dict if it is not present.

Reply:
Scratch that - we implemented this as an explicit registration: #3153